packages/editor/src/components/post-preview-button/test/index.js

```diff
@@ -1,188 +1,319 @@
  * Internal dependencies
  */
 import { PostPreviewButton } from '../';
 
 describe( 'PostPreviewButton', () => {
-	describe( 'setPreviewWindowLink()', () => {
-		it( 'should do nothing if there is no preview window', () => {
-			const url = 'https://wordpress.org';
-			const setter = jest.fn();
-			const wrapper = shallow( <PostPreviewButton /> );
+	const documentWrite = jest.fn();
+	const documentTitle = jest.fn();
+	const documentClose = jest.fn();
+	const setLocation = jest.fn();
 
-			wrapper.instance().setPreviewWindowLink( url );
+	beforeEach( () => {
+		global.open = jest.fn( () => ( {
+			focus: jest.fn(),
+			document: {
+				write: documentWrite,
+				close: documentClose,
+				get title() {},
+				set title( value ) {
+					documentTitle( value );
+				},
+			},
+			get location() {},
+			set location( value ) {
+				setLocation( value );
+			},
+		} ) );
+	} );
 
-			expect( setter ).not.toHaveBeenCalled();
-		} );
+	afterEach( () => {
+		global.open.mockRestore();
+		documentWrite.mockReset();
+		documentTitle.mockReset();
+		documentClose.mockReset();
+		setLocation.mockReset();
+	} );
 
-		it( 'set preview window location to url', () => {
-			const url = 'https://wordpress.org';
-			const setter = jest.fn();
-			const wrapper = shallow( <PostPreviewButton /> );
-			wrapper.instance().previewWindow = {
-				get location() {
-					return {
-						href: 'about:blank',
-					};
-				},
-				set location( value ) {
-					setter( value );
-				},
-			};
+	it( 'should render with `editor-post-preview` class if no `className` is specified.', () => {
+		render( <PostPreviewButton /> );
 
-			wrapper.instance().setPreviewWindowLink( url );
+		expect( screen.getByRole( 'button' ) ).toHaveClass(
+			'editor-post-preview'
+		);
+	} );
 
-			expect( setter ).toHaveBeenCalledWith( url );
-		} );
+	it( 'should render with a custom class and not `editor-post-preview` if `className` is specified.', () => {
+		render( <PostPreviewButton className="foo-bar" /> );
+
+		const button = screen.getByRole( 'button' );
+
+		expect( button ).toHaveClass( 'foo-bar' );
+		expect( button ).not.toHaveClass( 'editor-post-preview' );
 	} );
 
-	describe( 'getWindowTarget()', () => {
-		it( 'returns a string unique to the post id', () => {
-			const instance = new PostPreviewButton( {
-				postId: 1,
-			} );
+	it( 'should set `variant` to `tertiary` if no classname is specified.', () => {
+		render( <PostPreviewButton /> );
 
-			expect( instance.getWindowTarget() ).toBe( 'wp-preview-1' );
-		} );
+		expect( screen.getByRole( 'button' ) ).toHaveClass( 'is-tertiary' );
 	} );
 
-	describe( 'componentDidUpdate()', () => {
-		it( 'should change popup location if preview link is available', () => {
-			const wrapper = shallow(
-				<PostPreviewButton
-					postId={ 1 }
-					currentPostLink="https://wordpress.org/?p=1"
-					isSaveable
-					modified="2017-08-03T15:05:50"
-				/>
-			);
+	it( 'should not set `variant` if a custom classname is specified.', () => {
```

Comment:
Similarly to the previous comment:

```suggestion
	it( 'should render the button in its default variant if a custom classname is specified.', () => {
```